esphome/components/wifi/wifi_component_esp_idf.cpp

```diff
@@ -502,6 +505,12 @@ std::string format_ip4_addr(const esp_ip4_addr_t &ip) {
            uint8_t(ip.addr >> 24));
   return buf;
 }
+std::string format_ip6_addr(const esp_ip6_addr_t &ip) {
+  char buf[50];
+
+  snprintf(buf, sizeof(buf), IPV6STR, IPV62STR(ip));
+  return buf;
```

Comment:
You can't return `buf` here, buf is on the stack... Use the following to create a copy on the heap.

```suggestion
  return std::string(buf);
```

Reply:
Should I modify the `format_ip4_addr` in the same way?

Reply:
Uh yes, that seems to be a bug.

Reply:
Actually, it works fine because `buf` is a `char*` and the return type is `std::string`, so C++ does a copy initialization using the conversion constructor of `std::string`... but yes, being explicit about it is much better.

Reply:
Oh I see. Today I learned :smile: